resolwe/flow/executors/init_container.py

```diff
@@ -65,19 +67,36 @@ async def transfer_inputs(communicator: BaseCommunicator):
     inputs_connector.name = "inputs"
     inputs_connector.config["path"] = INPUTS_VOLUME
 
-    logger.debug(f"Inputs connector: {inputs_connector}")
-
     logger.debug("Transfering missing data.")
     response = await communicator.send_command(
         Message.command("get_inputs_no_shared_storage", "")
     )
+
+    # Group files by connectors. So we can transfer files from single connector
+    # in parallel. We could also transfer files belonging to different
+    # connectors in parallel but this could produce huge number of threads,
+    # since S3 uses multiple threads to transfer single file.
+    objects_to_transfer: dict[str, list[dict[str, str]]] = {}
+    for base_url, (connector_name, files) in response.message_data.items():
+        for file in files:
+            file["from_base_url"] = base_url
+            file["to_base_url"] = base_url
+        objects_to_transfer[connector_name] = (
```

Comment:
It would be nice to use `defaultdict` here.

Reply:
Sure.